### PR TITLE
Post sucess notifications (part 1)

### PR DESCRIPTION
**Pull request by mzorz**
This PR implements the basics of "success notifications" for Post uploads as described in this design piece:

screen shot 2017-10-18 at 10 25 11 am

This includes:
- look and feel and colors for action text buttons
- title and subtitle adjustments
- `Publish` quick action implementation for the successful notification for drafts uploaded.
- snackbars for success cases with new blue button
- making the success notification shown only when the app is not on the foreground

NOTES:
- kept the `draft uploaded` wording as opposed to `draft saved`, as I recall we were making this differentiation on purpose to assure the user this draft is on their site now (as opposed to having something locally saved only).
- the `CLOSE` quick action button is not implemented (we didn't have it, other apps don't use it, seems redundant to me)
- the 5 second duration is actually set to the more standard Android duration of `Snackbar.LENGTH_LONG` :)

Also in this PR:
- simplified success and error notification id handling in 08113cc

cc @nbradbury

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/WordPress.java [hunk 1]
```diff
@@ -736,6 +738,7 @@ private void stopActivityTransitionTimer() {
             }
 
             mIsInBackground = false;
+            sAppIsInTheBackground = false;
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/WordPress.java]: Do we really need two variables here which hold the same information?

**Comment by nbradbury** [comment]
> kept the draft uploaded wording as opposed to draft saved, as I recall we were making this differentiation on purpose to assure the user this draft is on their site now (as opposed to having something locally saved only).

> the CLOSE quick action button is not implemented (we didn't have it, other apps don't use it, seems redundant to me)

> the 5 second duration is actually set to the more standard Android duration of Snackbar.LENGTH_LONG :)

👍 on all of those.

File: WordPress/src/main/java/org/wordpress/android/WordPress.java (hunk 1 above)
Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/WordPress.java]: oops, addressed in ca2e6ef

File: WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadUtils.java [hunk 2]
```diff
@@ -245,6 +245,14 @@ public void onClick(View v) {
             } else {
                 int messageRes = post.isPage() ? R.string.page_published : R.string.post_published;
                 UploadUtils.showSnackbar(snackbarAttachView, messageRes);
+                UploadUtils.showSnackbarSuccessAction(snackbarAttachView, messageRes,
+                        R.string.button_view, new View.OnClickListener() {
+                            @Override
+                            public void onClick(View view) {
+                                // jump to Editor Preview mode to show this Post
+                                ActivityLauncher.viewPostPreviewForResult(activity, site, post);
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadUtils.java]: This is inconsistent with the `view` button in the post list, which uses `ActivityLauncher.browsePostOrPage()` to show the page in a WebView.

Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadUtils.java]: good observation! 🙇 addressed in c486c17

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**